### DISCUSSION TITLE
Type Generation of Views

### DISCUSSION THREAD

**Discussion by zephaniahong** [hasura/graphql-engine#8854] [Question] [answered]
Hi!
I am having an issue updating fields of a custom view using the Hasura Explorer vs on my own NextJs app which uses type generation to generate the types from Hasura.

I have a table `Event` with fields such as `id, userId, title, budget, description.` However, as I want to only give access of` budget, description` to the user which this event belongs to, I turned off the column select permission of `budget, description` on the `Event` table. I also created a View called `own_event` which has the fields `id, budget, description`. I then created an object relationship between `Event` and `own_event` by referencing the `userId` from the `Event` table to the `id` of the `own_event` view with the relationship name of `private_details`

The problem I'm facing is in 2 fold:
1. Using the Hasura Explorer, the `events_set_input` type does not contain the `private_details` field, but instead directly contains `description, budget`.
2. On my NextJs app, after generating the types, my `events_set_input` contains neither the `private_details` field nor the `description, budget` BUT the `events_insert_input` does indeed contain the private_details field.

I'm puzzled as to why the insert and set type are different even though I've double checked that the permissions I've set on my tables are all correct.

Thank you!

**Comment by adas98012** [comment, marked as answer]
Hi,
    Thanks for reaching out. I have created a Github issue for your query: 
    https://github.com/hasura/graphql-engine/issues/8868
     Please add reproduction steps (and relevant screenshots) there, to help our team troubleshoot the issue better.

**Reply by zephaniahong** [reply]
Will do! Thank you(:
